### ISSUE TITLE
Eleventy config API method to Set Data

### ISSUE THREAD

**Issue by kleinfreund** [11ty/eleventy#184]
This is possible since `v0.5.3`. For more information see [JavaScript Data Files](https://v0-5-3.11ty.io/docs/data-js/). Example:

**`_date/eleventy.js`**:
```js
module.exports = {
  'environment': process.env.ELEVENTY_ENV
};
```

**`example.liquid`**:
```liquid
<head>
  {% if eleventy.environment == 'development' %}

    <link rel="stylesheet" href="{{ '/css/styles.css' | url }}">

  {% else %}

    {% capture inline_css %}
      {% include layouts/styles.css %}
    {% endcapture %}
    <style>{{ inline_css | safe }}</style>

  {% endif %}
</head>
```

**Serve in development**:

```sh
ELEVENTY_ENV=development eleventy --serve
```

**Build for production**:

```sh
ELEVENTY_ENV=production eleventy
```

---

**Original feature request**:

I’d like to react to the current environment in my templates. This would enhance our capabilities to use Eleventy without additional build tools, e.g. for inlining and compress styles in production while referencing external CSS during development.

Preferably, an environment varialbe like `ELEVENTY_ENV` would be used for this (i.e. to match what users know from the likes of Jekyll [`JEKYLL_ENV`] and Hugo [`HUGO_ENV`]).

We have access to `process.env.ELEVENTY_ENV` in `.eleventy.js` out-of-the-box. Would it be reasonable to set data from within the config (maybe prefixed with `eleventy` as in `eleventy.environment`)? For comparison: With Jekyll, one is able to add data in the `_config.yml` file. Then, one can access it under the `site` key (e.g. `site.thing`).

**Comment by zachleat** [comment]
Work for this was started in `config-setdata` branch.

**Comment by kleinfreund** [comment]
Zach, are you planning to allow setting data via configuration as well? As far as the initial feature request is concerned, I’m very much satisfied. I don’t think it’s necessary to clutter the configuration file. Currently, this neatly separates configuration and data. If you think this is covered by [JavaScript Data Files](https://v0-5-3.11ty.io/docs/data-js/), this issue can be closed.

**Comment by zachleat** [comment]
You know, I’m not sure. I was going to punt on it for now and see how the JS Data Files plays out first.

I don’t quite think that the coverage is the same. Consider this use case that adds a new collection containing an array of tag strings:

https://github.com/11ty/eleventy-base-blog/commit/ef7783222dd48ffcab1acac0ba6df5a7272948be

You may want to set data based on other configuration API calls, maybe? A small use case but enough to leave this in the feature request queue, I think.

Gonna close this per https://github.com/11ty/eleventy/issues/184#issuecomment-409444230 to go back into the queue

**Comment by zachleat** [comment]
I’m going to remove this from the queue in favor of #245 (and a hefty overlap between the two ideas) for now. If someone comes up with a unique need for something like this, perhaps to work with plugins https://www.11ty.io/docs/plugins/ we can revisit.

**Comment by zachleat** [comment]
Going to pop this back into the queue after https://twitter.com/autiomaa/status/1199314595443363841

**Comment by zachleat** [comment]
Also https://twitter.com/khaled_garbaya/status/1227632474492108803

**Comment by zachleat** [comment]
For completeness this was shipped in v1.0.0 `eleventyConfig.addGlobalData` https://github.com/11ty/eleventy/pull/1280

https://www.11ty.dev/docs/data-global-custom/
